scripts/AcctThread.py

```
@@ -0,0 +1,269 @@
+       All responses are in json, so return json.'''
+    while True:
+        msg = ws.recv()
+        json_msg = json.loads(msg)
+        got_id = json_msg["id"]
+        if got_id == req_id:
+            return json_msg
+        print(
+            f"Unexpected websocket message id: {got_id}.  Expected {req_id}.")
+
+
+def print_account_info(ws, account_id):
+    '''Request account_info, print it, and return it as JSON'''
+    wsid = ws_id()
+
+    cmd = {
+        "id" : wsid,
+        "command" : "account_info",
+        "account" : account_id,
+        "strict" : True,
+        "ledger_index" : "validated"
+    }
+
+    ws.send(json.dumps(cmd))
+    json_msg = get_response(ws, wsid)
+
+    # Remove the websocket id to reduce noise.
+    json_msg.pop("id", None)
+    print(json.dumps(json_msg, indent=4))
+    return json_msg
+
+
+def print_tx(ws, txn_id):
+    '''Request tx by tnx_id, print it, and return the tx as JSON'''
+    wsid = ws_id()
+
+    cmd = {
+        "id": wsid,
+        "command": "tx",
+        "transaction": txn_id
+    }
+
+    ws.send(json.dumps(cmd))
+    json_msg = get_response(ws, wsid)
+
+    # Remove the websocket id from what we print to reduce noise.
+    json_msg.pop("id", None)
+    print(json.dumps(json_msg, indent=4))
+    return json_msg
+
+
+def get_prev_txn_id(json_msg, account_id):
+    '''Extract the PreviousTxnID for accountID given a transaction's metadata'''
+
+    # Handle any errors that might be in the response.
+    if "error" in json_msg:
+
+        # If the error is txnNotFound there there's a good chance
+        # the server does not have enough history.
+        if json_msg["error"] == "txnNotFound":
+            print("Transaction not found.",
+                  "Does your server have enough history? Unexpected stop.")
+            return ""
+
+        err = json_msg["error"]
+        if json_msg['error_message']:
+            err = json_msg['error_message']
```

I think this could be
```
err = json_msg['error_message'] or json_msg["error"]
```

---

Yup, that would work if my original code were correct.  Which it's not.  I had failed to check that error handling path.  The old code did not work because a missing key in a Dictionary throws a `KeyError` exception rather than return `None`.  Fortunately your nudge led me to actually _test_ that error condition.  So now I've changed the code to this:
```
        if "error_message" in json_msg:
            err = json_msg["error_message"]
        else:
            err = json_msg["error"] + "."
```
I think that's easier to read than the old code anyway.  And I also tested both legs this time 🙂.